src/main/java/sirius/biz/model/QueryController.java

```diff
@@ -165,6 +168,47 @@ private List<Property> determineVisibleProperties(EntityDescriptor descriptor) {
                          .toList();
     }
 
+    /**
+     * Provides a JSON based query facility.
+     * <p>
+     * Note that this is only available in TEST and DEV systems as it is only intended to be used for integration tests.
+     * <p>
+     * Just like the UI, this expects <tt>type</tt> to contain a Mixing descriptor name as produced by
+     * {@link sirius.db.mixing.Mixing#getNameForType(Class)} and <tt>query</tt> to contain a proper query
+     * e.g. "id:X". Additionally <tt>limit</tt> can specify how many entities are to be returned.
+     * <p>
+     * Note that this isn't a public API as it isn't available on production systems.
+     *
+     * @param webContext the request to handle
+     * @param output     the output to write the retrieved entities to
+     */
+    @Permission(TenantUserManager.PERMISSION_SYSTEM_ADMINISTRATOR)
```

Comment:
This would mean we would have to place username and API Token of an account with system admin authority in the systemtest repo. Isn't this kind of a security hazard? It's just the test and staging systems but maybe there is a better solution?

Reply:
Yeah, perhaps this API could rely on a config-key with a white-list of IPs or Networks masks.

Reply:
drone can provide (and securely store) "secrets" (which we also use for docker tasks etc.")....

Reply:
maybe jenkins can do the same (?)

Reply:
Maybe we can check that with Jonas

Reply:
Its noted in the ticket where we will implement such queries in our API tests.